src/progress-bar/progress-bar.test.tsx

```diff
@@ -0,0 +1,92 @@
+    percentage: 57,
+    style: {backgroundColor: "blue", completedColor: "yellow", trackColor: "orange"}
+  };
+
+  it("should render correctly", () => {
+    render(<ProgressBar {...defaultProgressBarProps} />);
+  });
+
+  it("should match snapshot", () => {
+    const tree = create(<ProgressBar {...defaultProgressBarProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    await testA11y(container);
+  });
+
+  it("should render children correctly", () => {
+    const progressBarContent = <span data-testid={"progress-bar.content"}>{"Test"}</span>;
+
+    const {getByTestId} = render(
+      <ProgressBar {...defaultProgressBarProps}>{progressBarContent}</ProgressBar>
+    );
+
+    expect(getByTestId(defaultProgressBarProps.testid!)).toContainElement(
+      getByTestId("progress-bar.content")
+    );
+  });
+
+  it("progress bar's background color should be backgroundColor", () => {
+    const {getByTestId} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    expect(getByTestId(defaultProgressBarProps.testid!)).toHaveStyle(
+      `background-color: ${defaultProgressBarProps.style.backgroundColor}`
+    );
+  });
+
+  it("the width of the progress bar track should be `percentage` percent", () => {
+    const {getByTestId} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    expect(getByTestId(`${defaultProgressBarProps.testid!}.track`)).toHaveStyle(
+      `width: ${defaultProgressBarProps.percentage}%`
+    );
+  });
+
+  it("if the percentage is less than 100, background color of the progress bar track should be trackColor", () => {
+    const {getByTestId} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    expect(getByTestId(`${defaultProgressBarProps.testid!}.track`)).toHaveStyle(
+      `background-color: ${defaultProgressBarProps.style.trackColor}`
+    );
+  });
+
+  it("if the percentage is greater or equal to 100, background color of the progress bar track should be completedColor", () => {
+    const {rerender, getByTestId} = render(
+      <ProgressBar {...defaultProgressBarProps} percentage={100} />
+    );
+
+    expect(getByTestId(`${defaultProgressBarProps.testid!}.track`)).toHaveStyle(
+      `background-color: ${defaultProgressBarProps.style.completedColor}`
+    );
+
+    rerender(<ProgressBar {...defaultProgressBarProps} percentage={120} />);
+
+    expect(getByTestId(`${defaultProgressBarProps.testid!}.track`)).toHaveStyle(
+      `background-color: ${defaultProgressBarProps.style.completedColor}`
+    );
+  });
+
+  it("if the percentage is greater or equal to 100,  width of the progress bar track should be 100%", () => {
+    const {getByTestId} = render(
+      <ProgressBar {...defaultProgressBarProps} percentage={120} />
+    );
+
+    expect(getByTestId(`${defaultProgressBarProps.testid!}.track`)).toHaveStyle(
+      "width: 100%"
+    );
+  });

```

Comment:
I think this can be checked within the block `Progress bar track has the right size`